### ISSUE TITLE
Alpine is not happy

### ISSUE THREAD

**Issue by gquintard** [varnishcache/varnish-cache#3565]
this is on `master`:

```
99b9eabd78e8:/workspace# make V=1
make  all-recursive
make[1]: Entering directory '/workspace'
Making all in include
make[2]: Entering directory '/workspace/include'
make  all-am
make[3]: Entering directory '/workspace/include'
make[3]: Leaving directory '/workspace/include'
make[2]: Leaving directory '/workspace/include'
Making all in lib
make[2]: Entering directory '/workspace/lib'
Making all in libvarnish
make[3]: Entering directory '/workspace/lib/libvarnish'
/bin/sh ../../libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I../..  -I../../include -I../../include   -DVARNISH_STATE_DIR='"/usr/local/var/varnish"'    -g -O2 -pthread -Wall -Werror -Wno-error=unused-result -fstack-protector-strong -Werror -Wall -Wcast-align -Wcast-qual -Wchar-subscripts -Wempty-body -Wextra -Wformat-y2k -Wformat-zero-length -Wmissing-field-initializers -Wmissing-prototypes -Wnested-externs -Wpointer-arith -Wpointer-sign -Wreturn-type -Wshadow -Wstrict-aliasing -Wstrict-prototypes -Wswitch -Wsystem-headers -Wunused-parameter -Wunused-result -Wwrite-strings -Wno-thread-safety -Wno-old-style-definition -Wno-sign-compare -Wno-implicit-fallthrough -Wno-builtin-requires-header -Wno-incomplete-setjmp-declaration -Wno-redundant-decls -Wno-missing-variable-declarations -Wno-missing-field-initializers -O0 -g -fno-inline -MT libvarnish_la-vcli_proto.lo -MD -MP -MF .deps/libvarnish_la-vcli_proto.Tpo -c -o libvarnish_la-vcli_proto.lo `test -f 'vcli_proto.c' || echo './'`vcli_proto.c
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../.. -I../../include -I../../include "-DVARNISH_STATE_DIR=\"/usr/local/var/varnish\"" -g -O2 -pthread -Wall -Werror -Wno-error=unused-result -fstack-protector-strong -Werror -Wall -Wcast-align -Wcast-qual -Wchar-subscripts -Wempty-body -Wextra -Wformat-y2k -Wformat-zero-length -Wmissing-field-initializers -Wmissing-prototypes -Wnested-externs -Wpointer-arith -Wpointer-sign -Wreturn-type -Wshadow -Wstrict-aliasing -Wstrict-prototypes -Wswitch -Wsystem-headers -Wunused-parameter -Wunused-result -Wwrite-strings -Wno-thread-safety -Wno-old-style-definition -Wno-sign-compare -Wno-implicit-fallthrough -Wno-builtin-requires-header -Wno-incomplete-setjmp-declaration -Wno-redundant-decls -Wno-missing-variable-declarations -Wno-missing-field-initializers -O0 -g -fno-inline -MT libvarnish_la-vcli_proto.lo -MD -MP -MF .deps/libvarnish_la-vcli_proto.Tpo -c vcli_proto.c  -fPIC -DPIC -o .libs/libvarnish_la-vcli_proto.o
In file included from /usr/include/sys/types.h:70,
                 from vcli_proto.c:33:
/usr/include/endian.h: In function '__bswap32':
/usr/include/endian.h:26:25: error: suggest parentheses around arithmetic in operand of '|' [-Werror=parentheses]
   26 |  return __x>>24 | __x>>8&0xff00 | __x<<8&0xff0000 | __x<<24;
      |                   ~~~~~~^~~~~~~
/usr/include/endian.h:26:41: error: suggest parentheses around arithmetic in operand of '|' [-Werror=parentheses]
   26 |  return __x>>24 | __x>>8&0xff00 | __x<<8&0xff0000 | __x<<24;
      |                                   ~~~~~~^~~~~~~~~
/usr/include/endian.h: In function '__bswap64':
/usr/include/endian.h:31:23: error: suggest parentheses around '+' inside '<<' [-Werror=parentheses]
   31 |  return __bswap32(__x)+0ULL<<32 | __bswap32(__x>>32);
      |         ~~~~~~~~~~~~~~^~~~~
vcli_proto.c: At top level:
cc1: note: unrecognized command-line option '-Wno-missing-variable-declarations' may have been intended to silence earlier diagnostics
cc1: note: unrecognized command-line option '-Wno-incomplete-setjmp-declaration' may have been intended to silence earlier diagnostics
cc1: note: unrecognized command-line option '-Wno-builtin-requires-header' may have been intended to silence earlier diagnostics
cc1: note: unrecognized command-line option '-Wno-thread-safety' may have been intended to silence earlier diagnostics
cc1: all warnings being treated as errors
make[3]: *** [Makefile:864: libvarnish_la-vcli_proto.lo] Error 1
make[3]: Leaving directory '/workspace/lib/libvarnish'
make[2]: *** [Makefile:418: all-recursive] Error 1
make[2]: Leaving directory '/workspace/lib'
make[1]: *** [Makefile:574: all-recursive] Error 1
make[1]: Leaving directory '/workspace'
make: *** [Makefile:459: all] Error 2
```

`ggc`:
```
99b9eabd78e8:/workspace# gcc --version
gcc (Alpine 10.2.1_pre1) 10.2.1 20201203
Copyright (C) 2020 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

and the `configure` command:
```
sudo -u varnish \
    ./configure \
	--quiet \
	--with-unwind \
	--enable-developer-warnings \
	--enable-debugging-symbols \
	--disable-stack-protector \
	--with-persistent-storage
```

could be an alpine specific issue, or the forwarning of troubles to come as a new `gcc` version rolls out

**Comment by gquintard** [comment]
thanks @Dridi, there are still those:

```
cc cannot -Wstring-plus-int

	cc: error: unrecognized command-line option '-Wstring-plus-int'

cc cannot -Wunused-parameters

	cc: error: unrecognized command-line option '-Wunused-parameters'; did you mean '-Wunused-parameter'?

make  dist-gzip am__post_remove_distdir='@:'
make[1]: Entering directory '/workspace'
make  distdir-am
make[2]: Entering directory '/workspace'
make  all-recursive
make[3]: Entering directory '/workspace'
Making all in include
make[4]: Entering directory '/workspace/include'
mkdir -p ../include/tbl
/usr/bin/python3.8 ../lib/libvcc/generate.py \
    .. ..
make  all-am
make[5]: Entering directory '/workspace/include'
  CC       vbm_test.o
  CCLD     vbm_test
make[5]: Leaving directory '/workspace/include'
make[4]: Leaving directory '/workspace/include'
Making all in lib
make[4]: Entering directory '/workspace/lib'
Making all in libvarnish
make[5]: Entering directory '/workspace/lib/libvarnish'
  CC       vjsn_test-vjsn.o
  CC       libvarnish_la-vbh.lo
  CC       libvarnish_la-vas.lo
  CC       libvarnish_la-vav.lo
  CC       libvarnish_la-vcli_proto.lo
  CC       libvarnish_la-vcli_serve.lo
  CC       libvarnish_la-vct.lo
  CC       libvarnish_la-venc.lo
  CC       libvarnish_la-version.lo
  CC       libvarnish_la-vev.lo
  CC       libvarnish_la-vfil.lo
  CC       libvarnish_la-vfl.lo
  CC       libvarnish_la-vin.lo
  CC       libvarnish_la-vjsn.lo
  CC       libvarnish_la-vlu.lo
In file included from /usr/include/pthread.h:30,
                 from vev.c:36:
/usr/include/sched.h: In function '__CPU_AND_S':
/usr/include/sched.h:110:1: error: cast discards 'const' qualifier from pointer target type []8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wcast-qual-Werror=cast-qual]8;;]
  110 | __CPU_op_func_S(AND, &)
      | ^~~~~~~~~~~~~~~
/usr/include/sched.h:110:1: error: cast discards 'const' qualifier from pointer target type []8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wcast-qual-Werror=cast-qual]8;;]
  110 | __CPU_op_func_S(AND, &)
      | ^~~~~~~~~~~~~~~
/usr/include/sched.h: In function '__CPU_OR_S':
/usr/include/sched.h:111:1: error: cast discards 'const' qualifier from pointer target type []8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wcast-qual-Werror=cast-qual]8;;]
  111 | __CPU_op_func_S(OR, |)
      | ^~~~~~~~~~~~~~~
/usr/include/sched.h:111:1: error: cast discards 'const' qualifier from pointer target type []8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wcast-qual-Werror=cast-qual]8;;]
  111 | __CPU_op_func_S(OR, |)
      | ^~~~~~~~~~~~~~~
/usr/include/sched.h: In function '__CPU_XOR_S':
/usr/include/sched.h:112:1: error: cast discards 'const' qualifier from pointer target type []8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wcast-qual-Werror=cast-qual]8;;]
  112 | __CPU_op_func_S(XOR, ^)
      | ^~~~~~~~~~~~~~~
/usr/include/sched.h:112:1: error: cast discards 'const' qualifier from pointer target type []8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wcast-qual-Werror=cast-qual]8;;]
  112 | __CPU_op_func_S(XOR, ^)
      | ^~~~~~~~~~~~~~~
  CC       libvarnish_la-vmb.lo
  CC       libvarnish_la-vnum.lo
vev.c: At top level:
cc1: note: unrecognized command-line option '-Wno-missing-variable-declarations' may have been intended to silence earlier diagnostics
cc1: note: unrecognized command-line option '-Wno-incomplete-setjmp-declaration' may have been intended to silence earlier diagnostics
cc1: note: unrecognized command-line option '-Wno-builtin-requires-header' may have been intended to silence earlier diagnostics
cc1: note: unrecognized command-line option '-Wno-thread-safety' may have been intended to silence earlier diagnostics
```

**Comment by gquintard** [comment]
fixed with 05780b634c40233dd2494a2c76a8dd8ba72d79f8

**Comment by Dridi** [comment]
https://github.com/varnishcache/varnish-cache/commit/f44f16cc4696e9e7f434dfd6d67b4e0447b4e97b

**Comment by gquintard** [comment]
still an issue:

```
autoreconf: Entering directory `.'
autoreconf: configure.ac: not using Gettext
autoreconf: running: aclocal -I m4 -I .
autoreconf: configure.ac: tracing
autoreconf: configure.ac: creating directory build-aux
autoreconf: running: libtoolize --copy
libtoolize: putting auxiliary files in AC_CONFIG_AUX_DIR, 'build-aux'.
libtoolize: copying file 'build-aux/ltmain.sh'
libtoolize: putting macros in AC_CONFIG_MACRO_DIRS, 'm4'.
libtoolize: copying file 'm4/libtool.m4'
libtoolize: copying file 'm4/ltoptions.m4'
libtoolize: copying file 'm4/ltsugar.m4'
libtoolize: copying file 'm4/ltversion.m4'
libtoolize: copying file 'm4/lt~obsolete.m4'
autoreconf: running: /usr/bin/autoconf
autoreconf: running: /usr/bin/autoheader
autoreconf: running: automake --add-missing --copy --no-force
configure.ac:10: installing 'build-aux/compile'
configure.ac:18: installing 'build-aux/config.guess'
configure.ac:18: installing 'build-aux/config.sub'
configure.ac:22: installing 'build-aux/install-sh'
configure.ac:22: installing 'build-aux/missing'
bin/varnishadm/Makefile.am: installing 'build-aux/depcomp'
parallel-tests: installing 'build-aux/test-driver'
autoreconf: Leaving directory `.'
configure: WARNING: dot not found - build will fail if svg files are out of date.
configure: WARNING: No system jemalloc found, using system malloc
cc cannot -Wstring-plus-int

	cc: error: unrecognized command-line option '-Wstring-plus-int'

cc cannot -Wunused-parameters

	cc: error: unrecognized command-line option '-Wunused-parameters'; did you mean '-Wunused-parameter'?

make  dist-gzip am__post_remove_distdir='@:'
make[1]: Entering directory '/workspace'
make  distdir-am
make[2]: Entering directory '/workspace'
make  all-recursive
make[3]: Entering directory '/workspace'
Making all in include
make[4]: Entering directory '/workspace/include'
mkdir -p ../include/tbl
/usr/bin/python3.8 ../lib/libvcc/generate.py \
    .. ..
make  all-am
make[5]: Entering directory '/workspace/include'
  CC       vbm_test.o
  CCLD     vbm_test
make[5]: Leaving directory '/workspace/include'
make[4]: Leaving directory '/workspace/include'
Making all in lib
make[4]: Entering directory '/workspace/lib'
Making all in libvarnish
make[5]: Entering directory '/workspace/lib/libvarnish'
  CC       vjsn_test-vjsn.o
  CC       libvarnish_la-vbh.lo
  CC       libvarnish_la-vas.lo
  CC       libvarnish_la-vav.lo
  CC       libvarnish_la-vcli_proto.lo
  CC       libvarnish_la-vcli_serve.lo
  CC       libvarnish_la-vct.lo
  CC       libvarnish_la-venc.lo
  CC       libvarnish_la-version.lo
  CC       libvarnish_la-vev.lo
  CC       libvarnish_la-vfil.lo
  CC       libvarnish_la-vfl.lo
  CC       libvarnish_la-vin.lo
In file included from /usr/include/pthread.h:30,
                 from vev.c:36:
/usr/include/sched.h: In function '__CPU_AND_S':
/usr/include/sched.h:110:1: error: cast discards 'const' qualifier from pointer target type []8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wcast-qual-Werror=cast-qual]8;;]
  110 | __CPU_op_func_S(AND, &)
      | ^~~~~~~~~~~~~~~
/usr/include/sched.h:110:1: error: cast discards 'const' qualifier from pointer target type []8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wcast-qual-Werror=cast-qual]8;;]
  110 | __CPU_op_func_S(AND, &)
      | ^~~~~~~~~~~~~~~
/usr/include/sched.h: In function '__CPU_OR_S':
/usr/include/sched.h:111:1: error: cast discards 'const' qualifier from pointer target type []8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wcast-qual-Werror=cast-qual]8;;]
  111 | __CPU_op_func_S(OR, |)
      | ^~~~~~~~~~~~~~~
/usr/include/sched.h:111:1: error: cast discards 'const' qualifier from pointer target type []8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wcast-qual-Werror=cast-qual]8;;]
  111 | __CPU_op_func_S(OR, |)
      | ^~~~~~~~~~~~~~~
/usr/include/sched.h: In function '__CPU_XOR_S':
/usr/include/sched.h:112:1: error: cast discards 'const' qualifier from pointer target type []8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wcast-qual-Werror=cast-qual]8;;]
  112 | __CPU_op_func_S(XOR, ^)
      | ^~~~~~~~~~~~~~~
/usr/include/sched.h:112:1: error: cast discards 'const' qualifier from pointer target type []8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wcast-qual-Werror=cast-qual]8;;]
  112 | __CPU_op_func_S(XOR, ^)
      | ^~~~~~~~~~~~~~~
  CC       libvarnish_la-vjsn.lo
  CC       libvarnish_la-vlu.lo
  CC       libvarnish_la-vmb.lo
  CC       libvarnish_la-vnum.lo
  CC       libvarnish_la-vpf.lo
  CC       libvarnish_la-vre.lo
vev.c: At top level:
cc1: note: unrecognized command-line option '-Wno-nullability-completeness' may have been intended to silence earlier diagnostics
cc1: note: unrecognized command-line option '-Wno-missing-variable-declarations' may have been intended to silence earlier diagnostics
cc1: note: unrecognized command-line option '-Wno-incomplete-setjmp-declaration' may have been intended to silence earlier diagnostics
cc1: note: unrecognized command-line option '-Wno-builtin-requires-header' may have been intended to silence earlier diagnostics
cc1: note: unrecognized command-line option '-Wno-thread-safety' may have been intended to silence earlier diagnostics
cc1: all warnings being treated as errors
make[5]: *** [Makefile:899: libvarnish_la-vev.lo] Error 1
```

**Comment by Dridi** [comment]
Of course, since I reverted your commit.

Let´s move the discussion in #3568 where we´ll try to iterate and find a proper fix,

**Comment by Dridi** [comment]
Giving up on alpine, it will run without developer warnings. The important part is that we don't don't wholesale remove crucial warnings for the benefit of a platform we don't actively support.

**Comment by Dridi** [comment]
I'm now fairly confident that I accidentally found the root cause and fixed it in #3641.
